qcodes/instrument_drivers/rohde_schwarz/HMC804x.py

```diff
@@ -0,0 +1,115 @@
+                           unit='V',
+                           vals=vals.Numbers(0, 32.050)
+                          )
+        self.add_parameter("set_current",
+                           label='Target current output',
+                           set_cmd="{} :SOURce:CURRent:LEVel:IMMediate:AMPLitude {}".format(
+                               select_cmd, '{}'),
+                           get_cmd="{} :SOURce:CURRent:LEVel:IMMediate:AMPLitude?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='A',
+                           vals=vals.Numbers(0.5e-3, self._parent.max_current)
+                           )
+        self.add_parameter('state',
+                           label='Output enabled',
+                           set_cmd='{} :OUTPut:CHANnel:STATe {}'.format(select_cmd, '{}'),
+                           get_cmd='{} :OUTPut:CHANnel:STATe?'.format(select_cmd),
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF')
+                           )
+        self.add_parameter("voltage",
+                           label='Measured voltage',
+                           get_cmd="{} :MEASure:SCALar:VOLTage:DC?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='V',
+                          )
+        self.add_parameter("current",
+                           label='Measured current',
+                           get_cmd="{} :MEASure:SCALar:CURRent:DC?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='A',
+                           )
+        self.add_parameter("power",
+                           label='Measured power',
+                           get_cmd="{} :MEASure:SCALar:POWer?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='W',
+                           )
+
+
+class RohdeSchwarzHMC804x(VisaInstrument):
+    """
+    This is the qcodes driver for the Rohde & Schwarz HMC804x Power Supply
+    """
+
+    _max_currents = {3: 3.0, 2: 5.0, 1: 10.0}
+
+    def __init__(self, name, address, num_channels, **kwargs):
+        super().__init__(name, address, **kwargs)
+
+        self.max_current = RohdeSchwarzHMC804x._max_currents[num_channels]
+
+        self.add_parameter('state',
+                           label='Output enabled',
+                           set_cmd='OUTPut:MASTer:STATe {}',
+                           get_cmd='OUTPut:MASTer:STATe?',
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF')
+                           )
+
+        # channel-specific parameters
+        channels = ChannelList(self, "SupplyChannel", RohdeSchwarzHMC804xChannel)
```

Comment:
The ChannelList should not be snapshotable, should it? The channels are also added individually below.

Reply:
Probably not. How can mark it a non-snapshotable?

Reply:
Give it the `snapshotable` keyword argument upon init.

From the `ChannelList` docstring:
```
        snapshotable (bool): Optionally disables taking of snapshots
            for a given channel list.  This is used when objects
            stored inside a channel list are accessible in multiple
            ways and should not be repeated in an instrument snapshot.
```